Backport of:

From 2e488845fbf7c6675ade42412f97f9abef091f33 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Mon, 4 Jul 2016 09:04:52 -0400
Subject: [PATCH] Prevent possible stack overflow

Prevent stack overflow by checking if string is null

(cherry picked from commit 6ea4d4e9eeac1e287bdbfda84f67fb1e50242fb5)

origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/6ea4d4e9eeac1e287bdbfda84f67fb1e50242fb5
bug-debian: https://bugs.debian.org/833812
---
 coders/clip.c      | 3 ++-
 coders/histogram.c | 3 ++-
 coders/mask.c      | 3 ++-
 coders/thumbnail.c | 3 ++-
 coders/vid.c       | 3 ++-
 5 files changed, 10 insertions(+), 5 deletions(-)

Index: imagemagick-6.6.9.7/coders/clip.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/clip.c	2016-11-15 15:56:13.862660141 -0500
+++ imagemagick-6.6.9.7/coders/clip.c	2016-11-15 15:56:52.907136886 -0500
@@ -169,7 +169,8 @@
   (void) CopyMagickString(clip_image->filename,image->filename,MaxTextExtent);
   write_info=CloneImageInfo(image_info);
   (void) SetImageInfo(write_info,1,&image->exception);
-  if (LocaleCompare(write_info->magick,"CLIP") == 0)
+  if ((*write_info->magick == '\0') ||
+      (LocaleCompare(write_info->magick,"CLIP") == 0))
     (void) FormatMagickString(clip_image->filename,MaxTextExtent,"miff:%s",
       write_info->filename);
   status=WriteImage(write_info,clip_image);
Index: imagemagick-6.6.9.7/coders/histogram.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/histogram.c	2016-11-15 15:56:13.862660141 -0500
+++ imagemagick-6.6.9.7/coders/histogram.c	2016-11-15 15:57:33.299630091 -0500
@@ -381,7 +381,8 @@
     MaxTextExtent);
   write_info=CloneImageInfo(image_info);
   (void) SetImageInfo(write_info,1,&image->exception);
-  if (LocaleCompare(write_info->magick,"HISTOGRAM") == 0)
+  if ((*write_info->magick == '\0') ||
+      (LocaleCompare(write_info->magick,"HISTOGRAM") == 0))
     (void) FormatMagickString(histogram_image->filename,MaxTextExtent,
       "miff:%s",write_info->filename);
   status=WriteImage(write_info,histogram_image);
Index: imagemagick-6.6.9.7/coders/thumbnail.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/thumbnail.c	2016-11-15 15:56:13.862660141 -0500
+++ imagemagick-6.6.9.7/coders/thumbnail.c	2016-11-15 15:57:53.435875961 -0500
@@ -208,7 +208,8 @@
     MaxTextExtent);
   write_info=CloneImageInfo(image_info);
   (void) SetImageInfo(write_info,1,&image->exception);
-  if (LocaleCompare(write_info->magick,"THUMBNAIL") == 0)
+  if ((*write_info->magick == '\0') ||
+      (LocaleCompare(write_info->magick,"THUMBNAIL") == 0))
     (void) FormatMagickString(thumbnail_image->filename,MaxTextExtent,
       "miff:%s",write_info->filename);
   status=WriteImage(write_info,thumbnail_image);
Index: imagemagick-6.6.9.7/coders/vid.c
===================================================================
--- imagemagick-6.6.9.7.orig/coders/vid.c	2016-11-15 15:56:13.862660141 -0500
+++ imagemagick-6.6.9.7/coders/vid.c	2016-11-15 15:58:16.912162613 -0500
@@ -332,7 +332,8 @@
     MaxTextExtent);
   write_info=CloneImageInfo(image_info);
   (void) SetImageInfo(write_info,1,&image->exception);
-  if (LocaleCompare(write_info->magick,"VID") == 0)
+  if ((*write_info->magick == '\0') ||
+      (LocaleCompare(write_info->magick,"VID") == 0))
     (void) FormatMagickString(montage_image->filename,MaxTextExtent,
       "miff:%s",write_info->filename);
   status=WriteImage(write_info,montage_image);
